File: timeago-parser/raw/java/CheckAll.java
import com.grack.nanojson.JsonObject;
import com.grack.nanojson.JsonParser;

import java.io.File;
import java.io.FileInputStream;
import java.util.Arrays;
import java.util.Map;

public class CheckAll {

    public static void main(String[] args) throws Exception {
        int SECONDS = 59, currentSeconds = 0;
        int MINUTES = 59, currentMinutes = 0;
        int HOURS = 23, currentHours = 0;
        int DAYS = 6, currentDays = 0;
        int WEEKS = 4, currentWeeks = 0;
        int MONTHS = 11, currentMonths = 0;
        int YEARS = 12, currentYears = 0;

        for (String name : Arrays.asList("seconds", "minutes", "hours", "days", "weeks", "months", "years")) {
            JsonObject object = JsonParser.object().from(new FileInputStream(new File("timeago-parser/raw/times/" + name + ".json")));

            for (Map.Entry<String, Object> entry : object.entrySet()) {
                JsonObject value = (JsonObject) entry.getValue();

                final int size = value.keySet().size();
                if (size >= 80) {
                    if (name.equals("seconds")) currentSeconds++;
                    if (name.equals("minutes")) currentMinutes++;
                    if (name.equals("hours")) currentHours++;
                    if (name.equals("days")) currentDays++;
                    if (name.equals("weeks")) currentWeeks++;
                    if (name.equals("months")) currentMonths++;
                    if (name.equals("years")) currentYears++;
                } else {
                    System.err.println("Missing some units in: " + name + " → " + entry.getKey() + " (current size = " + size + ")");
                }

                String number = entry.getKey().replaceAll("\\D", "");
                for (Map.Entry<String, Object> langsKeys : value.entrySet()) {
                    String lang = langsKeys.getKey();
                    String langValue = String.valueOf(langsKeys.getValue());

                    String langValueNumber = langValue.replaceAll("\\D", "");
                    if (!langValueNumber.equals(number)) {
                        final String msg = langValueNumber.isEmpty() ? "doesn't contain number" : "different number";
                        System.out.printf("%-20s[!]   %22s: %10s   = %s \n", entry.getKey(), msg, lang, langValue);
                    }
                }
            }
        }
        System.out.println("\n\nHow many:\n");

        if (currentSeconds == SECONDS) System.out.println("seconds: " + currentSeconds);
        else System.out.println("[!] missing seconds: " + currentSeconds);

        if (currentMinutes == MINUTES) System.out.println("minutes: " + currentMinutes);
        else System.out.println("[!] missing minutes: " + currentMinutes);

        if (currentHours == HOURS) System.out.println("hours: " + currentHours);
        else System.out.println("[!] missing hours: " + currentHours);

        if (currentDays == DAYS) System.out.println("days: " + currentDays);
        else System.out.println("[!] missing days: " + currentDays);

        if (currentWeeks == WEEKS) System.out.println("weeks: " + currentWeeks);
        else System.out.println("[!] missing weeks: " + currentWeeks);

        if (currentMonths == MONTHS) System.out.println("months: " + currentMonths);
        else System.out.println("[!] missing months: " + currentMonths);

        if (currentYears == YEARS) System.out.println("years: " + currentYears);
        else System.out.println("[!] missing years: " + currentYears);
    }
}

File: timeago-generator/src/main/java/org/schabi/newpipe/timeago_generator/CheckAll.kt
/*
 * SPDX-FileCopyrightText: 2026 NewPipe e.V. <https://newpipe-ev.de>
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package org.schabi.newpipe.timeago_generator

import kotlinx.io.buffered
import kotlinx.io.files.Path
import kotlinx.io.files.SystemFileSystem
import kotlinx.io.readString
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.jsonObject

fun main() {
    val maxSeconds = 59; var currentSeconds = 0
    val maxMinutes = 59; var currentMinutes = 0
    val maxHours = 23; var currentHours = 0
    val maxDays = 6; var currentDays = 0
    val maxWeeks = 4; var currentWeeks = 0
    val maxMonths = 11; var currentMonths = 0
    val maxYears = 12; var currentYears = 0

    val categories = listOf("seconds", "minutes", "hours", "days", "weeks", "months", "years")
    val regex = Regex("\\D")

    categories.forEach { name ->
        val filePath = Path("timeago-parser/raw/times/$name.json")
        val jsonObject = Json.parseToJsonElement(
            SystemFileSystem.source(filePath).buffered().readString()
        ).jsonObject

        jsonObject.map { (key, value) -> key to value.jsonObject }.forEach { (key, value) ->
            val size = value.keys.size
            if (size >= 80) {
                when (name) {
                    "seconds" -> currentSeconds++
                    "minutes" -> currentMinutes++
                    "hours" -> currentHours++
                    "days" -> currentDays++
                    "weeks" -> currentWeeks++
                    "months" -> currentMonths++
                    "years" -> currentYears++
                }
            } else {
                System.err.println("Missing some units in: $name → $key (current size = $size)")
            }

            val number = key.replace(regex, "")
            value.forEach { (lang, langValue) ->
                val langValue = langValue.toString()
                val langValueNumber = langValue.replace(regex, "")

                if (langValueNumber != number) {
                    val msg = if (langValueNumber.isEmpty()) "doesn't contain number" else "different number"
                    System.out.printf("%-20s[!]   %22s: %10s   = %s \n", key, msg, lang, langValue)
                }
            }
        }
    }

    println("\n\nHow many:\n")

    printResult("seconds", currentSeconds, maxSeconds)
    printResult("minutes", currentMinutes, maxMinutes)
    printResult("hours", currentHours, maxHours)
    printResult("days", currentDays, maxDays)
    printResult("weeks", currentWeeks, maxWeeks)
    printResult("months", currentMonths, maxMonths)
    printResult("years", currentYears, maxYears)
}

fun printResult(category: String, current: Int, expected: Int) {
    when (current) {
        expected -> println("$category: $current")
        else -> println("[!] missing $category: $current")
    }
}
